src/app/app.module.ts

```diff
@@ -1,30 +1,25 @@
 import { NgModule } from '@angular/core';
 import { BrowserModule } from '@angular/platform-browser';
 import { HttpClientModule } from '@angular/common/http';
-import { FormsModule } from '@angular/forms';
-
 import { AppRoutingModule } from './app-routing.module';
+import { StoreModule } from '@ngrx/store';
 import { AppComponent } from './app.component';
 import { MainComponent } from './main/main.component';
-import { UsersComponent } from './users/users.component';
 import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
-import { MatListModule } from '@angular/material/list';
-import { MatToolbarModule } from '@angular/material/toolbar';
-import { MatIconModule } from '@angular/material/icon';
-import { MatButtonModule } from '@angular/material/button';
+import { SharedModule } from './shared/shared.module';
+import { UsersModule } from './users/users.module';
+import * as fromApp from './store/app.reducer';
 
 @NgModule({
-  declarations: [AppComponent, MainComponent, UsersComponent],
+  declarations: [AppComponent, MainComponent],
   imports: [
     BrowserModule,
     AppRoutingModule,
     HttpClientModule,
-    FormsModule,
     BrowserAnimationsModule,
-    MatListModule,
-    MatToolbarModule,
-    MatIconModule,
-    MatButtonModule,
+    StoreModule.forRoot(fromApp.appReducer),
+    SharedModule,
```

Comment:
I think Shared module should not be part of app module.
Just include it only on Users Module.

Reply:
The thing is I need the Mat toolbar in app component